src/internal/oscal/generate.go

```diff
@@ -292,9 +299,49 @@ func generateModelTypes(obj map[string]interface{}, structId string, structName
 	return formattedStructName
 }
 
+// generateOscalComponentDocumentStruct generates the top-level struct for OSCAL data models.
+func generateOscalComponentDocumentStruct(oscalSchema map[string]interface{}, pkgName string, tags []string) string {
+	// Check if there is a top-level 'required' field.
+	if componentDefinitionField := oscalSchema["required"]; componentDefinitionField != nil {
+		// There is, so let's convert it to a string.
+		componentDefinitionString := fmt.Sprintf("%v", componentDefinitionField)
+
+		// Trim the [] braces away.
+		componentDefinition := strings.Trim(componentDefinitionString, "[]")
+
+		// Format the string from 'component-definition' to 'ComponentDefinition'.
+		formattedComponentDefinition := FmtFieldName(componentDefinition)
+
+		// Format struct tags.
+		tagList := make([]string, 0)
+		for _, tag := range tags {
+			tagList = append(tagList, fmt.Sprintf("%s:\"%s\"", tag, componentDefinition))
+		}
+
+		// Construct the struct string.
+		structString := fmt.Sprintf("type OscalComponentDocument struct {\n\t%s %s `%s`\n}\n", formattedComponentDefinition, formattedComponentDefinition, strings.Join(tagList, " "))
+
+		// Format the Go struct.
+		formattedStruct, err := format.Source([]byte(structString))
```

Comment:
Does formatting this single struct provide any value here if we format the structString after all the structs have been created too? (Line 162)

Reply:
The purpose of adding the formatting here was so that we could have predictable, structured output to test against in the unit tests. Glad to consider alternative ways of handling this

Reply:
sounds fair to me!

Reply:
@brandtkeller for my own clarification, are you wanting/expecting a solution that doesn't involve formatting the struct in this function? I'm not clear on which part of what I said sounded fair to you. Is the justification I provided for it being there fair? If so, is there anything else you see in this PR that would hinder approving/merging?

Reply:
No changes required - reviewing sporadically from my phone - current justification sounds reasonable. 

Will work to get a full review in ASAP

Reply:
Thank you! No rush. I was wanting to make sure I wasn't misunderstanding your comment and leaving desired changes unmade.